Source/AH/Mixins/Queue.lua

```diff
@@ -0,0 +1,30 @@
+Auctionator.AH.Queue = {}
+
+function Auctionator.AH.Queue:Init()
+  self.queue = {}
+  Auctionator.EventBus:Register(self, {
+    --Auctionator.AH.Events.Ready
+    Auctionator.AH.Events.SearchReady
+  })
+end
+
+function Auctionator.AH.Queue:Push(func)
+  table.insert(self.queue, func)
+
+  if Auctionator.AH.internals.throttling.searchReady then
+    self:_Pop()
+  end
+end
+
+function Auctionator.AH.Queue:ReceiveEvent(event)
+  print(event)
+  self:_Pop()
+end
+
+function Auctionator.AH.Queue:_Pop()
+  if #self.queue > 0 then
+    print("calling")
+    Auctionator.AH.internals.throttling:Call(self.queue[1])
+    table.remove(self.queue, 1)
```

Comment:
I feel like there's an opportunity for a race condition here?

Reply:
There would be if there were parallel threads, lua is all one thread, so we're ok